python/pyspark/sql/udf.py

```diff
@@ -133,6 +133,8 @@ def returnType(self):
                                 "UDFs: returnType must be a StructType.")
         elif self.evalType == PythonEvalType.SQL_GROUPED_AGG_PANDAS_UDF:
             try:
+                if isinstance(self._returnType_placeholder, StructType):
+                    raise TypeError
```

Comment:
Hm, @BryanCutler, sorry if I missed something but why do we throw a type error here?

Reply:
Grouped Agg UDFs don't allow a `StructType` return yet, and before relied on the call to `to_arrow_type` to raise an error. Since that no longer happens, need to raise it here.

Reply:
I see. Can you add some message while we're here? If this is going to be fixed soon, I am okay as is as well.

Reply:
Sure. I will try it as a followup, but a message for now will be good. I just noticed that grouped map wasn't catching a nested struct type, so I need to fix that anyway.

Reply:
Yea, either way sounds good to me. I'll leave it to you.